pkg/env/action.go

```diff
@@ -52,6 +53,10 @@ type action struct {
 func (a *action) runWithT(ctx context.Context, cfg *envconf.Config, t *testing.T) (context.Context, error) {
 	switch a.role {
 	case roleBeforeTest, roleAfterTest:
+		if cfg.DryRunMode() {
+			klog.V(2).Info("Skipping execution of roleBeforeTest and roleAfterTest due to framework being in dry-run mode")
```

Comment:
This message should probably also appear when the e2e-framework is starting to inform the user that before/after step functions will be skipped in dry-run mode.